File: test/corpus/ruby/duplicates.rb
# An exact clone pair: identical up to renamed identifiers, above the clone size
# floor.

# dendro-expect: duplicate
def total_price(items, tax)
  subtotal = 0
  i = 0
  while i < items.length
    subtotal = subtotal + items[i]
    i = i + 1
  end
  grand = subtotal + subtotal * tax
  rounded = round_value(grand)
  rounded
end

# dendro-expect: duplicate
def final_amount(rows, rate)
  accum = 0
  i = 0
  while i < rows.length
    accum = accum + rows[i]
    i = i + 1
  end
  whole = accum + accum * rate
  snapped = round_value(whole)
  snapped
end

File: test/corpus/java/duplicates.java
// An exact clone pair: identical up to renamed identifiers, above the clone size
// floor.

class DuplicateOps {
    // dendro-expect: duplicate
    int totalPrice(int[] items, int tax) {
        int subtotal = 0;
        for (int i = 0; i < items.length; i++) {
            subtotal = subtotal + items[i];
        }
        int grand = subtotal + subtotal * tax;
        int rounded = roundValue(grand);
        return rounded;
    }

    // dendro-expect: duplicate
    int finalAmount(int[] rows, int rate) {
        int accum = 0;
        for (int i = 0; i < rows.length; i++) {
            accum = accum + rows[i];
        }
        int whole = accum + accum * rate;
        int snapped = roundValue(whole);
        return snapped;
    }
}
